core/src/main/scala/cats/Foldable.scala

```diff
@@ -39,6 +39,32 @@ import simulacrum.typeclass
    * Eval[B]` to support laziness in a stack-safe way. Chained
    * computation should be performed via .map and .flatMap.
    *
+   * Because this fold is designed to handle a potentially infinite
+   * sequence, it does not construct the result by starting with the last element
+   * and working back towards the head of the sequence (as a regular right fold would).
```

Comment:
I think that this statement is a bit misleading. In general right-associated folds can support infinite collections while left-associated folds cannot, so I don't think that there would be any thing "regular" about a right fold that starts with the last element.